store.go

```diff
@@ -105,9 +111,11 @@ type ROBigDataStore interface {
 // A RWBigDataStore wraps up the read-write big-data related methods of the
 // various types of file-based lookaside stores that we implement.
 type RWBigDataStore interface {
-	// SetBigData stores a (potentially large) piece of data associated with this
-	// ID.
-	SetBigData(id, key string, data []byte) error
+	// SetBigData stores a (potentially large) piece of data associated
+	// with this ID.  If this is an image store, pass
+	// github.com/containers/image/manifest.Digest as digestManifest to
+	// allow ByDigest to find images by their correct digests.
+	SetBigData(id, key string, data []byte, digestManifest func([]byte) (digest.Digest, error)) error
```

Comment:
(Is there anything that benefits from the shared `{,RW,RO}BigDataStore` interfaces, so that it is worth it to force container store users to supply an unnecessary `digestManifest`? There may well be a good reason for this, I have no idea.

If nothing else, the `nil` parameters added for container stores below, even if correct and ultimately safe, are a bit ugly.)

Reply:
Hmm, at this point I don't think so.  I know at one point we wanted to refactor interfaces so that Store wasn't just one large interface of its own, but that hasn't really bought us much since then.

The logic for quietly backfilling in a value when there isn't one recorded is there for upgrade cases where the previously-used version didn't track item digests, but that was something we added in #109, so it could also be argued that we don't need to worry about those cases any more.

Dropping both of those would let us excise the need for the callback everywhere except for in `SetImageBigData()`, which might be worth it.

Reply:
(I’m not at all saying that the backfilling is unnecessary; I have no relevant knowledge and no opinion either way. I’ve only really been commenting on the internal API spreading the impact further than it seems necessary, only based on my limited view of how c/storage is used.)

Reply:
Okay, dropped the backfill logic.  Kept the interfaces for grouping, but made the interfaces implemented by image and container stores different.